components/range_cache_memory_engine/src/range_manager.rs

```diff
@@ -336,189 +577,247 @@ impl RangeManager {
     //
     // For 2, this is caused by some special operations such as merge and delete
     // range. So, conservatively, we evict all ranges overlap with it.
-    pub(crate) fn evict_range(
+    pub(crate) fn evict_region(
         &mut self,
-        evict_range: &CacheRange,
+        evict_region: &Region,
         evict_reason: EvictReason,
-    ) -> Vec<CacheRange> {
+    ) -> Vec<Region> {
         info!(
-            "try to evict range";
-            "evict_range" => ?evict_range,
+            "try to evict region";
+            "evict_region" => ?evict_region,
+            "reason" => ?evict_reason,
         );
 
-        // cancel loading ranges overlapped with `evict_range`
-        self.pending_ranges_loading_data
-            .iter_mut()
-            .for_each(|(r, _, canceled)| {
-                if evict_range.overlaps(r) {
-                    info!(
-                        "evict range that overlaps with loading range";
-                        "evicted_range" => ?evict_range,
-                        "overlapped_range" => ?r,
-                    );
-                    *canceled = true;
+        if let Some(meta) = self.regions.get(&evict_region.id) {
+            // if epoch not changed, no need to do range scan.
+            if meta.region.get_region_epoch().version == evict_region.get_region_epoch().version {
+                if let Some(region) =
+                    self.do_evict_region(evict_region.id, evict_region, evict_reason)
+                {
+                    return vec![region];
                 }
-            });
-
-        let mut overlapped_ranges = vec![];
-        for r in self.ranges.keys() {
-            if r.contains_range(evict_range) {
-                if self.evict_within_range(evict_range, &r.clone(), evict_reason) {
-                    return vec![evict_range.clone()];
-                } else {
-                    return vec![];
-                }
-            } else if r.overlaps(evict_range) {
-                overlapped_ranges.push(r.clone());
             }
         }
 
```

Comment:
Could you add a log for recording failed eviction? Eg, epoch mistmatch or region not found.